evap/staff/templates/staff_semester_view.html

```diff
@@ -292,50 +359,49 @@ <h3>
-                                <a role="button" class="btn btn-sm btn-light fas fa-chart-bar icon-yellow" data-filter="fa-chart-bar icon-yellow"
-                                    data-toggle="tooltip" data-placement="top" title="{% trans 'Results not yet published' %}"></a>
-                                <a role="button" class="btn btn-sm btn-light fas fa-chart-bar icon-green" data-filter="fa-chart-bar icon-green"
-                                    data-toggle="tooltip" data-placement="top" title="{% trans 'Results published' %}"></a>
-                            </div>
-                        </div>`
-                    );
-                    $('[data-filter]').click( function(e) {
-                        $(e.target).toggleClass("active");
-                        $("[data-filter].active").not(e.target).removeClass("active");
+                    $('#course-filter-buttons [data-filter]').click(function(e) {
+                        // toggle active states and refresh table
+                        var button = $(e.target).closest("a");
+                        button.toggleClass("active");
+                        $("[data-filter].active").not(button).removeClass("active");
                         table.draw();
-                    } );
-                    activateTooltips('.filter-toolbar');
-                },
+
+                        // store which button was selected
+                        localStorage.removeItem('course-filter_{{ semester.id }}');
+                        if($("[data-filter].active").length)
+                        {
+                            localStorage['course-filter_{{ semester.id }}'] = $("[data-filter].active")[0].dataset.filter;
+                        }
+                    });
+                    // helper function to count number of lines with certain icon
+                    countIcons = function(obj) {
+                        var count = 0;
+                        $.each([1, 2, 3, 4], function(index, value) {
+                            count += table.column(value).data()
+                                .filter(function(value, index) {
+                                    return value.includes(obj.dataset.filter);
+                                })
+                                .count();
+                        });
+                        return count;
+                    };
+                    // add counts to each filter button
+                    $("#course-filter-buttons [data-filter]").each(function(i, obj) {
+                        var count = countIcons(obj);
+                        if (count > 0) {
+                            $('<span class="badge badge-pill badge-btn">' + count + '</span>').appendTo(obj);
+                        }
+                        else {
+                            $('<span class="badge badge-pill badge-btn-zero">' + count + '</span>').appendTo(obj);
+                        }
+                    });
```

Comment:
maybe swap this with the `.click` part above so the localstorage stuff is located together?